### ISSUE TITLE
Pure JAX Random Forest Implementation for NAO Brainwave Control - #320

### ISSUE THREAD

**Issue by Esma-Ali** [3C-SCSU/Avatar#377]
### **Summary**
The previous implementation used `sklearn.ensemble.RandomForestClassifier`, violating the requirement for a **pure JAX** model.  
This update introduces a **custom Random Forest built entirely in JAX**, using `jax.numpy` for computation and `jax.random` for reproducible RNG.  
The model now achieves **0.9996 accuracy and F1 macro**, with clean artifact generation and reproducible results.

---
### **Steps to Reproduce**
```bash
git fetch origin feat/jax-rf-320-v3
git checkout feat/jax-rf-320-v3
python -m venv .venv
.venv\Scripts\activate
pip install -r requirements.txt
python deep-learning-prediction/JAX/combine_brainwaves.py
python deep-learning-prediction/JAX/train_jax_rf.py
```
---
### **Expected**
- Combined dataset: `data/brainwaves.csv` + `data/brainwaves.labels.json`

- Training output:
```bash
accuracy = 0.9996
f1_macro = 0.9996
```
- Artifacts:
- [ ] `metrics.json`
- [ ] `confusion_matrix.png`
-No sklearn dependencies.
---
### Actual
- Model trains fully and reproducibly
- 0.9996 accuracy confirmed
- Artifacts generated under `JAX/artifacts/`
- No `scikit-learn` imports found
- Deterministic results via `jax.random.key()`
---
### **Fix**

**combine_brainwaves.py**
→ Merges EEG data from all folders and generates label mappings.

**jax_rf.py**
→ Implements full JAX Random Forest with custom tree building (`gini_impurity`, `best_split`, `grow`) using `jax.numpy` and `jax.random`.

**train_jax_rf.py**
→ Loads data, trains model, computes metrics, and saves artifacts.

---
### **How to Test**
```bash
python deep-learning-prediction/JAX/combine_brainwaves.py
python deep-learning-prediction/JAX/train_jax_rf.py
```
**Expected**
```bash
accuracy = 0.9996
f1_macro = 0.9996
```
Artifacts appear in `deep-learning-prediction/JAX/artifacts/`.
---
### **Follow-ups**
- Add `jax.jit` for faster training
- Integrate ONNX export for NAO6
- Add k-fold CV and CLI wrapper
---
### **Links**

- **Issue:** #320
- **Branch:** `feat/jax-rf-320-v3`
- **Files Changed:**
- combine_brainwaves.py
  - jax_rf.py
  - train_jax_rf.py
  - artifacts/*
